### ISSUE TITLE
3D CNN - Unknown data_format value: NDHWC in function 'getDataLayout'

### ISSUE THREAD

**Issue by moberweger** [opencv/opencv#15161]
##### System information (version)
- OpenCV == 4.1.0
- python == 2.7.12
- Operating System / Platform == Ubuntu 16.04 x86_64
- compiler == gcc 7.4.0
- tensorflow == 1.14.0

##### Detailed description
Loading a 3D CNN model from tensorflow to be run in OpenCV triggers an error when running `cv2.dnn.readNetFromTensorflow`. The cause of the error is related to 3D convolutions and Batch Normalization.

The code raises the following error:
`cv2.error: OpenCV(4.1.0) /io/opencv/modules/dnn/src/tensorflow/tf_importer.cpp:262: error: (-212:Parsing error) Unknown data_format value: NDHWC in function 'getDataLayout'`

##### Steps to reproduce
The following python script can be used to trigger the error.

```.py
#!/usr/bin/env python
import numpy
import cv2
import tensorflow as tf
from tensorflow.tools.graph_transforms import TransformGraph

if __name__ == '__main__':
    batch_size = 4
    input_shape = [batch_size, 32, 32, 32, 1]
    features = tf.placeholder(tf.float32, input_shape, name='input')

    features0 = tf.layers.conv3d(inputs=features, filters=8, kernel_size=[3, 3, 3], padding='same')
    in1 = tf.contrib.layers.batch_norm(features0, data_format='NHWC', center=True, scale=True, is_training=False)
    in2 = tf.nn.relu(in1)

    print([n.name for n in tf.get_default_graph().as_graph_def().node])

    with tf.Session() as sess:
        sess.run(tf.global_variables_initializer())
        constant_graph = tf.graph_util.convert_variables_to_constants(sess, sess.graph.as_graph_def(), ['Relu'])
        tf.train.write_graph(constant_graph, "", "graph_final.pb", as_text=False)

    # export
    with tf.gfile.FastGFile("graph_final.pb") as f:
        graph_def = tf.GraphDef()
        graph_def.ParseFromString(f.read())
        graph_def = TransformGraph(graph_def, ['input'], ['Relu'], ['strip_unused_nodes'])
        with tf.gfile.FastGFile('saved_model.pb', 'wb') as f:
            f.write(graph_def.SerializeToString())

    # read model
    cvNet = cv2.dnn.readNetFromTensorflow('./saved_model.pb')
    img = numpy.zeros((32, 32, 32, 1), dtype='float32')
    cvNet.setInput(img[None, ...].repeat(batch_size, axis=0))
    cvOut = cvNet.forward()
    print len(cvOut)
 ```

**Comment by dkurt** [comment]
>  cv2.error: OpenCV(4.1.0) /io/opencv/modules/dnn/src/tensorflow/tf_importer.cpp:262: error: (-212:Parsing error) Unknown data_format value: NDHWC in function 'getDataLayout'

According to [changelog](https://github.com/opencv/opencv/wiki/ChangeLog#version411) 3D CNNs are supported starts from OpenCV 4.1.1

**Comment by moberweger** [comment]
Thanks @dkurt for the hint to 4.1.1. I tried with the latest version from master (4.1.1-dev) and still I get an error. The same code from above fails now with the message:
`cv2.error: OpenCV(4.1.1-dev) /opencv/modules/dnn/src/layers/convolution_layer.cpp:307: error: (-215:Assertion failed) ngroups > 0 && inpCn % ngroups == 0 && outCn % ngroups == 0 in function 'getMemoryShapes'`
